tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+def test_positive_sync_plan_with_hammer_defaults(request, sat_maintain):
+    """Verify that sync plan is disabled and enabled with hammer defaults set.
+
+    :id: b25734c8-470f-4cad-bc56-5c0f75aa7499
+
+    :steps:
+        1. Setup hammer on system with defaults set
+        2. Run satellite-maintain advanced procedure run sync-plans-disable
+        3. Run satellite-maintain advanced procedure run sync-plans-enable
+
+    :expectedresults: sync plans should get disabled and enabled.
+    """
+    sat_maintain.cli.Defaults.add({'param-name': 'organization_id', 'param-value': 1})
+
+    result = sat_maintain.cli.Advanced.run_sync_plans_disable()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Advanced.run_sync_plans_enable()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    @request.addfinalizer
+    def _finalize():
+        sat_maintain.cli.Defaults.delete({'param-name': 'organization_id'})
+
+
+def test_positive_satellite_repositories_setup(sat_maintain):
+    """Verify that all required repositories gets enabled.
+
+    :id: e32fee2d-2a1f-40ed-9f94-515f75511c5a
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run repositories-setup --version 6.y
+
+    :BZ: 1684730, 1869731
+
+    :expectedresults: Required Satellite repositories for install/upgrade should get enabled
+    """
+    supported_versions = ['6.11', '6.12']
+    for ver in supported_versions:
+        result = sat_maintain.cli.Advanced.run_repositories_setup(options={'version': ver})
+        assert 'FAIL' not in result.stdout
+        assert result.status == 0
+        result = sat_maintain.execute('yum repolist')
+        for repo in sat_repos[ver]:
+            assert repo in result.stdout
+
+    # 6.13 till not GA
+    result = sat_maintain.cli.Advanced.run_repositories_setup(options={'version': '6.13'})
+    assert 'FAIL' in result.stdout
+    assert result.status == 1
+    for repo in sat_repos['6.13']:
+        assert repo in result.stdout
+
+    # Verify that all required beta repositories gets enabled
+    # maintain beta repo is unavailable for EL8 https://bugzilla.redhat.com/show_bug.cgi?id=2106750
+    sat_beta_repo = ['satellite-6-beta-for-rhel-8-x86_64-rpms'] + common_repos
+    missing_beta_el8_repos = ['satellite-maintenance-6-beta-for-rhel-8-x86_64-rpms']
+    result = sat_maintain.cli.Advanced.run_repositories_setup(
+        options={'version': '6.12'}, env_var='FOREMAN_MAINTAIN_USE_BETA=1'
+    )
+    assert 'FAIL' in result.stdout
+    assert result.status != 0
+    for repo in missing_beta_el8_repos:
+        assert f"Error: '{repo}' does not match a valid repository ID" in result.stdout
+    result = sat_maintain.execute('yum repolist')
+    for repo in sat_beta_repo:
+        assert repo in result.stdout
```

Comment:
Could we separate this test for common repos and beta sat repo ?

Reply:
No, anyway we're not getting any beta repositories nowadays

Reply:
Then why do we need this to verify it even in this test ?

Reply:
No, its good to handle this here, beta releases are not there for el8 based sat yet, but it will be there in future, and its already handled, so I wouldn't block on this